### DISCUSSION TITLE
How to make responsive UI by statement?

### DISCUSSION THREAD

**Discussion by roberts-erglis** [slint-ui/slint#6253] [Q&A] [answered]
In documentation only shows on callbacks thing but how then on just widgets/elements itself?
I tried if-else on Rectangle element:
`Rectangle {`
`  if (parent.height > 500px;) {`
`    height: parent.height - 100px;`
`  } else {`
`    height: 500px;`
`  }`
`}`
**Is there correct syntax or it's not yet in Slint?**

**Comment by yanshay** [comment, marked as answer]
You can use :
` height: (parent.height > 500px) ? (parent.height -100px) : 500px; `

**Reply by roberts-erglis** [reply]
Yes, did not came on mind that I could use shorten version of if-else. Thanks!
_(There is missing "px" in condition)_

**Reply by yanshay** [reply]
> (There is missing "px" in condition)

This is what happens when trying to code inside GitHub Comment Box 😄 
I fixed original in case someone copy/paste.

**Comment by felipetesc** [comment]
or ...
add your code inside a function 

```slint
...
//place outside the component using it
// give a name to the instance of the component you want to make the changes

//here I called component
function getHeight( ) -> length {
    if (component.height > 500px) {
      return parent.height - 100px;
    } else {
      return 500px;
    }
}
...
height : getHeight(); // call when it is needed 
...
```
https://releases.slint.dev/1.7.2/docs/slint/src/language/builtins/elements.html#id20

**Reply by roberts-erglis** [reply]
Also works for clean code and more complex situation. Thanks man!
Changed that instead _component_ but _root_, becasue it did not work outside for some reason...
